src/polycap-photon.c

```diff
@@ -0,0 +1,274 @@
+		//obtain selected capillary indices
+		i_capx = round((photon->start_coords.x / description->profile->ext[0]) * n_shells);
+		i_capy = round((photon->start_coords.y / (description->profile->ext[0]*sin(M_PI/3.))) * n_shells);
+		//define some shell size parameters
+		cap_unita[0] = description->profile->ext[0] / (n_shells);
+		cap_unita[1] = 0.;
+		cap_unitb[0] = cap_unita[0] * cos(M_PI/3.);
+		cap_unitb[1] = cap_unita[0] * sin(M_PI/3.);
+		//convert indexed capillary centre to coordinates
+		capx_0 = i_capx * cap_unita[0] + i_capy * cap_unitb[0];
+		capy_0 = i_capx * cap_unita[1] + i_capy * cap_unitb[1];
+	}
+
+	//define selected capillary axis X and Y coordinates
+	//NOTE: Assuming polycap centre coordinates are X=0,Y=0 with respect to photon->start_coords
+	cap_x = malloc(sizeof(double)*(description->profile->nmax+1));
+	if(cap_x == NULL){
+		printf("Could not allocate cap_x memory.\n");
+		exit(1);
+	}
+	cap_y = malloc(sizeof(double)*(description->profile->nmax+1));
+	if(cap_y == NULL){
+		printf("Could not allocate cap_y memory.\n");
+		exit(1);
+	}
+	
+	for(i=0; i<=description->profile->nmax; i++){
+		cap_x[i] = description->profile->ext[i] * capx_0 / description->profile->ext[0];
+		cap_y[i] = description->profile->ext[i] * capy_0 / description->profile->ext[0];
+	}
+
+	//calculate initial photon weight based on capillary channel effective solid angle.
+	//Mathematically, this is the cos of the angle between photon propagation and polycapillary-to-photonsource axis
+	weight = polycap_scalar(photon->start_direction,central_axis);
+	for(i=0; i<photon->n_energies; i++){
+		photon->weight[i] = photon->weight[i] * weight;
+	}
+
+	
+	//polycap_capil_trace should be ran description->profile->nmax at most,
+	//which means it essentially reflected once every known capillary coordinate
+	for(i=0; i<=description->profile->nmax; i++){ 
+		iesc = polycap_capil_trace(ix, photon, description, cap_x, cap_y);
+		if(iesc != 0){ //as long as iesc = 0 photon is still reflecting in capillary
+		//iesc == -2, which means this photon has reached its final point (weight[0] <1e-4)
+			//in old program a new photon is simulated at this point
+		//alternatively, iesc can be 1 due to not finding intersection point, as the photon reached the end of the capillary
+			break;
+		}
+	}
+
+	if(iesc == -2){
+		return -1; //return -1 if photon did not reach end of capillary
+	} else {
+		return 0; //if photon reached end of capillary, return 0
+	}
+}
+
+//===========================================
+// get exit coordinates
+polycap_vector3 polycap_photon_get_exit_coords(polycap_photon *photon)
+{
+	return photon->exit_coords;
+}
+
+//===========================================
+// get exit direction
+polycap_vector3 polycap_photon_get_exit_direction(polycap_photon *photon)
+{
+	return photon->exit_direction;
+}
+
+//===========================================
+// get exit electric vector
+polycap_vector3 polycap_photon_get_exit_electric_vector(polycap_photon *photon)
```

Comment:
Wat is trouwens uw mening over de elektrische vector? Doet dat er toe?

Reply:
Mja, heb er ook al ferm over zitten nadenken. Ik weet niet direct of het de mate in reflectivity echt zou aanpassen, maar kvroeg me vooral af of de richting van dat veld ook zou mee variëren. Zoja lijkt het inderdaad interessant om dit mee te geven.
Hoe dan ook, kging het er eens met Laszlo over hebben, hij gaat veel directer kunnen zeggen wat en of er invloeden zijn enzo :) In ieder geval lijkt het me niet zo'n mega grote aanpassing om in te voegen, gewoon even zorgvuldig nadenken over juiste richting van vectoren enzo om de juiste factoren s- en p-polarisatie toe te wijzen. Voorlopig wou ik het echter zonder doen, omdat het dan gemakkelijker te vergelijken is met de oude code, om te zien of alles naar behoren werkt :)

Reply:
Na overleg met Laszlo:
Het is in ieder geval iets waar hij indertijd ook in geïnteresseerd was om echt te simuleren wat de verschillen zouden zijn (met polarisatie-afhankelijke reflecties etc), maar in praktijk heeft hij nooit echt bewijs gezien dat het veel zou schelen. Bvb de reflectiviteit zal nauwelijks veranderen doordat je tientallen of zelfs honderden reflecties hebt in het capillair, die alles toch wat uitmiddelen. Ook de electric field vector zal nauwelijks veranderen doordat je onder zeer kleine hoeken reflecteert.
Dus het lijkt me iets wat ik zeker later in de code wil steken, maar momenteel doe ik even voort zoals het is om zeker te zijn dat het werkt etc zoals voorheen :)

Reply:
Ok fair enough... Misschien voor in een latere release dan...